### ISSUE TITLE
Setting childWhenDragging to new Container() when Dragging a DragTarget asserts.

### ISSUE THREAD

**Issue by apwilson** [flutter/flutter#6087]
```
16:13:37.710: I/flutter : ══╡ EXCEPTION CAUGHT BY GESTURE LIBRARY ╞═══════════════════════════════════════════════════════════
16:13:37.713: I/flutter : The following assertion was thrown while routing a pointer event:
16:13:37.716: I/flutter : setState() called after dispose(): _DragTargetState<int>(848060255; _StateLifecycle.defunct; not
16:13:37.719: I/flutter : mounted)
16:13:37.722: I/flutter : This error happens if you call setState() on a State object for a widget that no longer appears in
16:13:37.725: I/flutter : the widget tree (e.g., whose parent widget no longer includes the widget in its build). This error
16:13:37.728: I/flutter : can occur when code calls setState() from a timer or an animation callback. The preferred solution
16:13:37.730: I/flutter : is to cancel the timer or stop listening to the animation in the dispose() callback. Another
16:13:37.733: I/flutter : solution is to check the "mounted" property of this object before calling setState() to ensure the
16:13:37.736: I/flutter : object is still in the tree.
16:13:37.740: I/flutter : This error might indicate a memory leak if setState() is being called because another object is
16:13:37.742: I/flutter : retaining a reference to this State object after it has been removed from the tree. To avoid memory
16:13:37.744: I/flutter : leaks, consider breaking the reference to this object during dispose().
16:13:37.747: I/flutter :
16:13:37.750: I/flutter : When the exception was thrown, this was the stack:
16:13:37.753: I/flutter : #0      State.setState.<anonymous closure> (package:flutter/src/widgets/framework.dart:818:9)
16:13:37.755: I/flutter : #2      State.setState (package:flutter/src/widgets/framework.dart:816:12)
16:13:37.758: I/flutter : #3      _DragTargetState.didLeave (package:flutter/src/widgets/drag_target.dart:319:5)
16:13:37.760: I/flutter : #4      _DragAvatar._leaveAllEntered (package:flutter/src/widgets/drag_target.dart:455:26)
16:13:37.763: I/flutter : #5      _DragAvatar.updateDrag (package:flutter/src/widgets/drag_target.dart:428:5)
16:13:37.766: I/flutter : #6      _DragAvatar.update (package:flutter/src/widgets/drag_target.dart:389:5)
16:13:37.769: I/flutter : #7      MultiDragPointerState._move (package:flutter/src/gestures/multidrag.dart:84:15)
16:13:37.772: I/flutter : #8      MultiDragGestureRecognizer._handleEvent (package:flutter/src/gestures/multidrag.dart:208:13)
16:13:37.774: I/flutter : #9      PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:73:12)
16:13:37.777: I/flutter : #10     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:101:11)
16:13:37.780: I/flutter : #11     BindingBase&SchedulerBinding&GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:133:19)
16:13:37.783: I/flutter : #12     BindingBase&SchedulerBinding&GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:111:22)
16:13:37.786: I/flutter : #13     BindingBase&SchedulerBinding&GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:92:7)
16:13:37.788: I/flutter : #14     BindingBase&SchedulerBinding&GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:51:7)
16:13:37.791: I/flutter : #15     BindingBase&SchedulerBinding&GestureBinding._handlePointerPacket (package:flutter/src/gestures/binding.dart:44:5)
16:13:37.794: I/flutter : #16     _dispatchPointerPacket (file:///b/build/slave/Linux_Engine/build/src/flutter/lib/ui/hooks.dart:43)
16:13:37.797: I/flutter : (elided one frame from class _AssertionError)
16:13:37.800: I/flutter :
16:13:37.803: I/flutter : Event:
16:13:37.805: I/flutter :   PointerMoveEvent(Point(952.0, 355.5))
16:13:37.808: I/flutter : ════════════════════════════════════════════════════════════════════════════════════════════════════

```

**Comment by apwilson** [comment]
I think what's happening is the Draggable is a DragTarget and it enters itself before the "childWhenDragging" takes over upon drag start.

**Comment by abarth** [comment]
Looks like this was fixed by 42f773c327c2f4e9b85164a3cc0c480259f53d85 as well.
